pkg/storage/sql/sql.go

```diff
@@ -7,47 +7,81 @@ import (
 )
 
 type sql struct {
-	client *gorm.DB
+	Client *gorm.DB
 }
 
+type Handler func(value interface{}) error
+
 func (db *sql) Get(value interface{}) error {
-	err := db.client.Last(value).Error
+	err := db.Client.Last(value).Error
 	if err != nil {
 		return errors.E(err, util.ErrNotFound)
 	}
 	return nil
 }
 
+func (db *sql) Find(out interface{}, where ...interface{}) error {
+	err := db.Client.Find(out, where...).Error
+	if err != nil {
+		return errors.E(err, util.ErrNotFound)
+	}
+	return nil
+}
+
+func (db *sql) CreateOrUpdate(value interface{}) error {
+	if db.Client.NewRecord(value) {
+		return db.Add(value)
+	}
+	return db.Update(value)
+}
+
 func (db *sql) Update(value interface{}) error {
-	if db.client.NewRecord(value) {
+	if db.Client.NewRecord(value) {
```

Comment:
Could you do this function as proxy for toMany?

Reply:
do you say to write this verification only one time?

I think not, because is some cases we need to verify exist and other cases if don't exist